src/cavendish_particle_tracks/_stereoshift_dialog.py

```diff
@@ -81,51 +88,50 @@ def __init__(self, parent: "ParticleTracksWidget"):
         self.layout().addWidget(
-            QLabel("Select Reference / Fiducial"), 0, 0, 1, 2
+            QLabel("Select Reference / Fiducial"), 1, 0, 1, 2
         )
-        self.layout().addWidget(QLabel("Fiducial coordinates"), 1, 0, 1, 2)
-        self.layout().addWidget(self.cbf1, 0, 2)
+        self.layout().addWidget(self.cbf1, 1, 2)
+        self.layout().addWidget(QLabel("Fiducial coordinates"), 2, 0, 1, 2)
         for i, widget in enumerate(
             [lviewf1, self.textboxes[0], lviewf2, self.textboxes[1]]
         ):
-            self.layout().addWidget(widget, i // 2 + 2, i % 2 + 1)
-        self.layout().addWidget(QLabel("Point coordinates"), 4, 0, 1, 2)
+            self.layout().addWidget(widget, i // 2 + 3, i % 2 + 1)
+        self.layout().addWidget(QLabel("Point coordinates"), 5, 0, 1, 2)
         for i, widget in enumerate(
             [lviewb1, self.textboxes[2], lviewb2, self.textboxes[3]]
         ):
-            self.layout().addWidget(widget, i // 2 + 5, i % 2 + 1)
+            self.layout().addWidget(widget, i // 2 + 6, i % 2 + 1)
 
         self.layout().addWidget(bss, 7, 0, 1, 3)
 
         self.layout().addWidget(
             self.label_stereoshift,
-            8,
+            9,
             0,
             1,
             3,
         )
         # self.layout().addWidget(self.table, 9, 0, 1, 3)
-        self.layout().addWidget(QLabel("Fiducial shift"), 9, 1)
-        self.layout().addWidget(self.tshift_fiducial, 9, 2)
-        self.layout().addWidget(QLabel("Point shift"), 10, 1)
-        self.layout().addWidget(self.tshift_point, 10, 2)
-        self.layout().addWidget(QLabel("Ratio"), 11, 1)
-        self.layout().addWidget(self.tstereoshift, 11, 2)
-        self.layout().addWidget(QLabel("Point depth (cm)"), 12, 1)
-        self.layout().addWidget(self.tdepth, 12, 2)
-        self.layout().addWidget(bap, 13, 0, 1, 3)
-        self.layout().addWidget(self.buttonBox, 14, 0, 1, 3)
+        self.layout().addWidget(QLabel("Fiducial shift"), 10, 1)
+        self.layout().addWidget(self.tshift_fiducial, 10, 2)
+        self.layout().addWidget(QLabel("Point shift"), 11, 1)
+        self.layout().addWidget(self.tshift_point, 11, 2)
+        self.layout().addWidget(QLabel("Ratio"), 12, 1)
+        self.layout().addWidget(self.tstereoshift, 12, 2)
+        self.layout().addWidget(QLabel("Point depth (cm)"), 13, 1)
+        self.layout().addWidget(self.tdepth, 13, 2)
+        self.layout().addWidget(bap, 14, 0, 1, 3)
+        self.layout().addWidget(self.buttonBox, 15, 0, 1, 3)
 
         # Setup points layer
         self.cal_layer = self._setup_stereoshift_layer()
 
         # Stereoshift related parameters
-        self.shift_fiducial = 0.0
-        self.shift_point = 0.0
-        self.point_stereoshift = 0.0
-        self.point_depth = -1.0
-        self.spoints = []  # type: ignore
+        self.stereoshift_info: StereoshiftInfo = StereoshiftInfo()
+        self.stereoshift_info.name = "origin_vertex"
```

Comment:
Perhaps better to have two objects here? I think that might make your assignment to the ParticleDecay a little simpler because you won't need to deep copy.

Can we give the name in the constructor also?

(Deliberately not a code suggestion) Something like this:
```py
self.stereoshift_info_origin = StereoshiftInfo(name="origin_vertex")
self.stereoshift_info_decay = StereoshiftInfo(name="decay_vertex")
```

Pretty sure you can also remove the type hint, as you're initialising with the explicit constructor. `mypy` should be happy with this.

Reply:
?